great_expectations/core/data_context_key.py

```diff
@@ -94,7 +90,7 @@ def to_tuple(self) -> Tuple[str, Optional[str]]:
         """
         See parent `DataContextKey.to_tuple` for more information.
         """
-        return (self._resource_type, self._resource_name)
+        return (self._resource_type, self._resource_name or "")
```

Comment:
We require string values in the tuple per some downstream code - this logic is replicated in the `GeCloudStoreBackend`'s `GeCloudIdentifier` key